[Runtime] Move UI assets to Stride.UI + include StrideUILibrary as root

manio143 opened the pull request:
# PR Details

## Description

Move UI assets from `Stride.Engine` to `Stride.UI` - this is non-breaking as assets are identified by GUID.
Include `StrideUILibrary.sduilib` as a root asset for the `Stride.Engine` package.

![image](https://github.com/stride3d/stride/assets/10709060/3e6671ff-9750-4a79-b8db-b6275c68f49e)

I've verified nothing is broken by running a few existing samples and creating a new UI page which references the UI library and assets being moved around.

## Motivation and Context
Currently when creating a `UIPage` in code one has to provide all the details for controls, like fonts, sprites, etc. Those assets are available, just not included. This change allows to do the following:

```csharp
var uiLibrary = Content.Load<UILibrary>("StrideUILibrary");
var button = uiLibrary.InstantiateElement<Button>("Button");
```
And get a fully displayable button.

I understand that not everyone might want to include the default UI elements in their project, but it's just additional 128KB (less than the default splash screen) and allows to easily create dev UI in code without having to link the font, button sprites, etc. in the asset.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have built and run the editor to try this change out.

_Doesn't **require** documentation change, but:_
It would be nice to then mention this in https://doc.stride3d.net/latest/en/manual/ui/ui-libraries.html

manio143 committed:
Move UI assets to the Stride.UI package and include StrideUILibrary as root

xen2 commented:
I need to take a quick look if OK.
In any case, the place it should be is `Stride.UI` package rather than `Stride.Engine`.

froce committed:
Rework `Profiler` to support multiple subscribers and use Channels internally.

- Channels simplified the addition of multiple subscribers and should be a big perf win, when collecting events from many threads.
- Adds thread information to `ProfilingEvent` and `ProfilingState`
- Adapted `GameProfilingSystem` to the changes

froce committed:
Add some trace profiling scopes.

manio143 committed:
Make a few changes around viewport grid axis rotation and hiding

manio143 committed:
Introduce an enum to avoid allocating the nullable object

garychia committed:
Fix the default scene settings data

mattiascibien committed:
Fix documentation error on Int2

manio143 committed:
Merge pull request #1585 from mattiascibien/fix_int2_docs

Fix documentation error on Int2

Jklawreszuk committed:
Use LZ4 as external dependency

manio143 committed:
Merge pull request #1566 from Jklawreszuk/lz4-package

Use LZ4 as external dependency

Jklawreszuk committed:
Remove settings.json and update gitignore

manio143 committed:
Merge pull request #1586 from Jklawreszuk/vscode-gitignore

Remove settings.json and update gitignore

Eideren committed:
[Rendering] Addressing PVS report - SDL (#1546)

Eideren committed:
[Physics] Update WorldMatrix when forcing transformation on PhysicsComponent (#1406)

xen2 committed:
Merge remote-tracking branch 'arturoc/openxr-dx11-master' (closes #1483)

mattiascibien committed:
Fix #1466: adding scripts by dragging

Jklawreszuk committed:
Cleanup Windows Phone refs

Jklawreszuk committed:
Cleanup Silverlight refs

Jklawreszuk committed:
Cleanup WinStore refs

Jklawreszuk committed:
Cleanup Win8 & Win10 refs

Jklawreszuk committed:
Cleanup NET Portable,MicroFramework,Client

Jklawreszuk committed:
Remove WinStore content for VRSandbox sample

azeno committed:
[Graphics] If possible use flip model for swap chain

https://devblogs.microsoft.com/directx/dxgi-flip-model

Jklawreszuk committed:
Move launcher prerequisites validation to separate file

xen2 committed:
Keep local copy of signclient to avoid issues with teamcity

Jklawreszuk committed:
Launcher - Remove unnecessary namespaces, format and simplify code

manio143 committed:
Merge pull request #1605 from Jklawreszuk/launcher-refactor

[Launcher] Code refactor

garychia committed:
Merge pull request #1 from manio143/viewportgrid_changes

Viewportgrid changes

manio143 committed:
Merge pull request #1516 from garychia/GridsInMultiplePlanes

Enable multiple selections of the axes of grids to be drawn

manio143 committed:
Document what's in the Core folder

manio143 committed:
Merge pull request #1610 from manio143/document_core_folder

[Docs] Add a Readme with information on what's in the Core folder

manio143 committed:
Merge pull request #1588 from mattiascibien/fix_adding_scripts_by_dragging

Fix #1466: adding scripts by dragging

manio143 committed:
Update NuGet version to cover most .NET 6 SDK users for MSBuild issue

manio143 committed:
Add debug information & remove legacy code

manio143 committed:
Merge pull request #1613 from manio143/fix_script_editor

Update NuGet dependencies to 6.4.0 (temporary fix for ScriptEditor issues)

dependabot[bot] committed:
Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

Bumps [SSH.NET](https://github.com/sshnet/SSH.NET) from 2016.1.0 to 2020.0.2.
- [Release notes](https://github.com/sshnet/SSH.NET/releases)
- [Commits](https://github.com/sshnet/SSH.NET/compare/2016.1.0...2020.0.2)

---
updated-dependencies:
- dependency-name: SSH.NET
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Doprez committed:
fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1618 from Doprez/fix-begginner-errors

fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1614 from stride3d/dependabot/nuget/sources/engine/Stride.Assets/SSH.NET-2020.0.2

Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

manio143 committed:
Merge pull request #1428 from vvvv/dev/azeno/native-libs-use-runtimes-folder

[Build] Native libraries get copied to runtimes/RID/native folder

azeno committed:
[Graphics] Fixes Vsync off when using new flip model

azeno committed:
[Build] Fixes native libraries not copied to output folder of unit tests

For example the Stride.Navigation.Tests didn't contain the necessary libnavigation.dll in its runtimes folder.

xen2 committed:
Merge pull request #1632 from stride3d/dev/azeno/unit-test-native-lib-load-fix

[Build] Fixes native libraries not copied to output folder of unit tests

azeno committed:
[Graphics] Adds try/catch blocks to flip model and tearing support queries to support older OS

azeno committed:
[Graphics] Fixes crash when flip model is not supported

The `AllowTearing` flag must not be set in that case

Doprez committed:
rebased branch

manio143 committed:
Merge pull request #1643 from Doprez/change-msbuild

rebased branch

manio143 committed:
Merge pull request #1591 from Jklawreszuk/cleanup-legacy-platforms

Cleanup legacy platforms

NicusorN5 committed:
Possible fix for the animation importer crashing

`FbxTime::GetOneFrameValue(FbxTime::eFrames1000);` returns the smallest value, so there's the highest animation resolution possible(?)

Jklawreszuk committed:
[Core] Remove some obsolete code

NicusorN5 committed:
Update AnimationConverter.h

NicusorN5 committed:
Also check negative values

manio143 committed:
Merge pull request #1653 from NicusorN5/master

Possible fix for the animation importer crashing

NicusorN5 committed:
Update the forum link with Github discussions link

Doprez committed:
added null check

manio143 committed:
Merge pull request #1656 from Doprez/fix-NativeDevice-causing-crash

fix Crash when NativeDeviceContext is null

azeno committed:
[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package as well as used Silk version

This broke with the merge of PR #1428 because these lines of code where added after #1428 was created.

xen2 committed:
Merge pull request #1660 from vvvv/fix-openxr-load-location

[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package

manio143 committed:
Merge pull request #1654 from Jklawreszuk/core-obsoletes

[Core] Remove some obsolete code

Jklawreszuk committed:
Fix XML warnings (#1678)

Jklawreszuk committed:
[deps] - Remove unused ATITC folder (#1668)

Jklawreszuk committed:
[deps] - Remove unused HtmlRenderer (#1670)

Jklawreszuk committed:
[deps] - Remove unused ExpresionBlend Sdk (#1669)

Jklawreszuk committed:
Remove VisualStudio reattach deadcode (#1677)

Jklawreszuk committed:
Fix XML documentation (#1680)

IXLLEGACYIXL committed:
fix argument reference

IXLLEGACYIXL committed:
remvoe reference to non existing class that doesnt get referenced

IXLLEGACYIXL committed:
fix light to renderlight

IXLLEGACYIXL committed:
blend layer context is the only class that actually sets and evaluates the enum values

IXLLEGACYIXL committed:
remove duplicate through inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
fix wrong reference to draweffect parameters

IXLLEGACYIXL committed:
Merge branch 'FixXmlDocs' of https://github.com/NexStandard/stride into FixXmlDocs

IXLLEGACYIXL committed:
fix wrong comment. False class in  comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
fix wrong interface link

IXLLEGACYIXL committed:
remove duplicate

IXLLEGACYIXL committed:
fix wrong reference

IXLLEGACYIXL committed:
fix resourcegrouplayout reference

IXLLEGACYIXL committed:
fix vertex comment

IXLLEGACYIXL committed:
wrong param name, its still a weird name as it was previous a sorter i assume.

IXLLEGACYIXL committed:
ParticleSorter is now ParticleList from previous interface renaming

manio143 committed:
Merge pull request #1682 from NexStandard/FixXmlDocs

[Docs] Fix xml docs

IXLLEGACYIXL committed:
[Docs] Fix xml docs (#1683)

Ethereal77 committed:
Renamed build scripts in solution files and removed missing ones (#1696)

azeno committed:
[Reflection] Fixes loading of mixed mode assemblies in .NET Core (#1691)

It seems .NET Core behaves slightly different in this regard, throwing a different kind of exception, so previous code no longer applies, even though it's the same assembly being loaded (irrKlang.NET4.dll)

Jklawreszuk committed:
[deps] - Remove unused SSH (#1687)

Jklawreszuk committed:
[deps] Remove unused libgit2 dll's (#1686)

Jklawreszuk committed:
[deps] - Remove unused Irony (#1685)

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

Eideren committed:
Merge remote-tracking branch 'origin/master' into tasks_clarification

Eideren committed:
Revert "Merge remote-tracking branch 'origin/master' into tasks_clarification" (#1701)

This reverts commit 06dc98ab5e742d9ac34c5422403e25cd278e2e00.

Jklawreszuk committed:
Remove native code from stride.core project (#1688)

* Remove native code from stride.core project

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

Eideren committed:
Add dot between major and minor

Jklawreszuk committed:
Separate VisualStudio projects from main solution (#1684)

Jklawreszuk committed:
Fix edge case collision behavior with BoxColliderShape by creating Box2D shape when is2D (#1708)

IXLLEGACYIXL committed:
[GameStudio] Fix sub projects reloading - changing the comparison project related and not UPath related (#1704)

* fix creation and renaming for sub projects

bro-these-are-all-taken committed:
[Rendering] Vulkan Fullscreen Fix (#1709)

* Enabling swapchain recreation in SwapChainGraphicsPresenter.Vulkan.cs so that changes in resolution at runtime don't stop the game from rendering to the screen.

IXLLEGACYIXL committed:
[WIP] Replace NativeStream with BinaryWriter (#1710)

* Rename false Comments of Nativestream to Stream

* Replace any calls to BinaryReader/BinaryWriter

* Remove native stream IO

* remove using directives of deleted IO namespace

* Remove obsolete comment

* replace nativestream comment and block writing with copyto as it allready exists

Ethereal77 committed:
[Rendering] Fix missing Utilities.CopyMemory for other graphics APIs (#1697)

* Fix Direct3D 12 uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix OpenGL uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix Vulkan uses of Utilities.CopyMemory

* Also some minor cleanup

* Add a section to Stride.props where users can set design-time graphics API

* Fix more usages of CopyMemory and ClearMemory

* Also remove some helper methods that copied memory in favor of new built-in ones

* Comment modifications based on feedback

* Fix a deserialization issue introduced in commit 189e96980e96d6180e18a8f7e9466945ff9ecbe8

Eideren committed:
Docs: Readme, update Community Projects link (#1720)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: add Jklawreszuk as a contributor for code (#1727)

* docs: update README.md [skip ci]

* docs: update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

Jklawreszuk committed:
Change Assimp to Silk.NET.Assimp implementation (#1719)

* Changed Assimp binding to Silk.Net.Assimp

* Converted Stride.Importer.Assimp to a C# library and changed assimp binding to Silk.Net.Assimp
* Converted Stride.Importer.Common to a C# library as this did not contain any native code

allcontributors[bot] committed:
docs: add Ethereal77 as a contributor for code (#1729)

allcontributors[bot] committed:
docs: add tamamutu as a contributor for code (#1731)

allcontributors[bot] committed:
docs: add IXLLEGACYIXL as a contributor for code (#1732)

allcontributors[bot] committed:
docs: add arturoc as a contributor for code (#1733)

allcontributors[bot] committed:
docs: add garychia as a contributor for code (#1734)

allcontributors[bot] committed:
docs: add NicusorN5 as a contributor for code (#1735)

allcontributors[bot] committed:
docs: add mattiascibien as a contributor for code (#1736)

Jklawreszuk committed:
Update README badges (#1737)

Jklawreszuk committed:
[deps] - Remove unused assimp source code

Eideren committed:
[Samples] Bump samples version (#1756)

VaclavElias committed:
Readme.md update - Improving readme based on the community feedback (#1739)

* Removed duplicated badge

* Missing lfs added

arturoc committed:
Fix DX11 shared textures

Right now creating a texture from a native texture with D3D11_RESOURCE_MISC_SHARED_NTHANDLE  will fail unless it is also D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX. This combination is recomended but not mandatory acording to the specification: https://learn.microsoft.com/en-us/windows/win32/api/d3d11/ne-d3d11-d3d11_resource_misc_flag

Eideren committed:
[System] Fix binary path location when the dlls are run from an unexpected executable

xen2 committed:
Fix VSIX build

xen2 committed:
Assimp: preload assimp library from Ultz.Native.Assimp so that it can be found when actually needed (fixes #1765)

xen2 committed:
Removed UWP from README.md

VaclavElias committed:
[Service] Xml comments improvements (#1768)

Aggror committed:
Update README.md

I build stride source from scratch today but building worked only partially.

I was have SDK 7 installed, but was still missing SDK 6.0.

Also the Git LFS clone command is deprecated and buildin into Git clone

tebjan committed:
[Shader] Adds sRGB and Gamma conversions to ColorUtility.sdsl (#1744)

Co-authored-by: tebjan <tebjan@gmail.com>

Basewq committed:
[Graphics] Fix expected image size for OpenGL textures. (#1786)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

manio143 committed:
[Profiler] Refactor Profiler and submit metrics to .NET Diagnostics API (#1774)

Eideren committed:
[Asset] Show warning message when importing assets that are not inside of the resource folder (#1781)

manio143 committed:
Merge pull request #1746 from Jklawreszuk/assimp-code

[deps] - Remove unused assimp source code

manio143 committed:
Merge pull request #1702 from Eideren/clarification_2

[Build] Show useful error message and directions for users lacking requirements to run Stride.Core.Tasks

Eideren committed:
[Project] Show custom message for users with outdated dotnet version (#1767)

Eideren committed:
[Editor] PropertyGrid, re-implement member sorting by declaration/token (#1790)

Doprez committed:
[Sample] Fix SpriteFont sample, typo in service fetch (#1798)

SVNMLR committed:
[Editor] Add dynamic snapping (#1801)

Implements a dynamic snapping used while holding down a key on manipulating (rotating/moving/scaling) an object/entity

chkw-rks committed:
[Readme] Update logo for dark and light theme (#1587)

Jklawreszuk committed:
[deps] - Remove unused WindowsAPICodePack (#1747)

IXLLEGACYIXL committed:
[Editor] Repair NPE introduced in #1781 (#1804)

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

tebjan committed:
Adds Awesome Stride to readme

Schossi committed:
parse numbers in NumericTextBox using CurrentCulture
enables numpad usage for cultures that use comma for decimals

VaclavElias committed:
Update README.md - Bullet point fixed in Prerequisites (#1808)

Schossi committed:
parse numbers in NumericTextBox using broader culture settings
enables numpad usage for cultures that use comma for decimals

cNoNim committed:
[Editor] Fix some xaml errors (#1822)

Eideren committed:
[Graphics] Dx11: Fix ResizeBuffer when swap chain is under flip model and changing format (#1814)

VaclavElias committed:
[Core] Basic support for F# and VB Project Types (#1821)

SeleDreams committed:
[Editor] Added the ability to copy imported assets automatically to the Resources dir (#1827)

Doprez committed:
[Samples] Add more documentation for beginner tutorial (#1830)

MeharDT committed:
[Editor] Check to see if the .NET 6.0 SDK is Installed When Launching Stride (#1809)

Jklawreszuk committed:
Remove unused MonoDevelop.Debugger (#1748)

Jklawreszuk committed:
[deps] - Remove unused monolinker (#1749)

Eideren commented:
Did you end up taking a look at this one @xen2 ? 
@manio143 still needs to come back for that change request though

SVNMLR committed:
[Editor] Add new Theme & UI fixes/refinements (#1807)

MeharDT committed:
[iOS] .NET 6 Template (#1833)

- Updated iOS template .csproj to support .NET Core/6.0, Stride will no longer throw an error when generating iOS projects

cNoNim committed:
Fix compilation of AssemblyProcessor to netstandard2.0

cNoNim committed:
Update Mono.Cecil version to suppress version mismatch

manio143 commented:
Hmm, currently the Stride.UI doesn't contain any assets, but I think it should be possible to move the StrideUILibrary into that project while preserving the Guid identifier in order to not cause a breaking change for users.

Eideren committed:
[Physics] Pre and Post simulation tick event (#1831)

Eideren committed:
[Editor] Log viewer shows stacktrace by default (#1844)

xen2 committed:
Merge pull request #1840 from cNoNim/fix-assemblyprocessor

Fix compilation of AssemblyProcessor to netstandard2.0

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

JeromyWalsh committed:
Replaced 'xcopy' Exec task with the more universal Copy task to support OS's which do not have xcopy.

manio143 committed:
Merge pull request #1854 from JeromyWalsh/master

Replaced 'xcopy' Exec task with the more universal Copy task to suppo…

Jklawreszuk committed:
[Editor] Translations fix (#1717)

Commit:
feat(theme): improve editor tooltip legibility in Light Steel Blue theme

The Light Steel Blue theme had issues with legibility in editor tooltips, where it used black font on an almost-black background. This change addresses this problem by updating the theme styles.

Fixes #1773

manio143 committed:
Merge pull request #1863 from acastrodev/topic/acastrodev/1191

fix(theme): Improve editor tooltip legibility in Light Steel Blue theme

Eideren committed:
[Serialization][Editor] Reference type support on interface fields and properties (#1842)

Eideren committed:
[Core] Fix StringComparison across the project (#1846)

Eideren committed:
[Physics] Change simulation substeping behavior (#1849)

Eideren committed:
[Physics] Add ray test flags (#1850)

JeromyWalsh committed:
[Build] Fix wildcard expansion with AssemblyProcessor + PropertyGroup for environment variables (#1860)

parhamgholami committed:
[Editor] Always show directional light arrow (#1865)

manio143 committed:
[Roslyn] Add base setup for a Roslyn integration into Stride.Core (#1847)

* Set up analyzer project to be included in every project

* Create basic structure

* Add Tests project

* Remove generator, fixup tests

* Remove test analyzer

* Add compiler services to unit tests projects

xen2 committed:
Readd VisualStudio.Commands project (they were removed from #1684 but we want to build the Commands project as part of runtime release, even when we don't build/release VSIX package)

xen2 committed:
Fix VSPackage build

MeharDT committed:
[Editor] Fix exception with the Privacy Policy link in the Crash Reporter (#1878)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

froce committed:
Small Profiler fixes

- Flip ordering of Profiling results, to sort from longest to shortest.
- Only record `End` events in the Histogram, Begin/Mark events don't have a duration.

froce committed:
Add `ChromeTracingProfileWriter`

to output profiling events in 'chrome://tracing' JSON format, which can be analyzed by ui.perfetto.dev and other tools

froce committed:
Fix tests

Basewq committed:
[Graphics] Fix missing OpenGLES texture formats. (#1898)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

IXLLEGACYIXL committed:
[Docs] Typo in summary for DataMemberMode.Assign (#1900)

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

SVNMLR committed:
[Editor] Improve Cameracontrol in Editor (#1879)

* [Editor] Allow panning while Alt-Keys are beeing pressed

* Change: Remove check for altIsDown

* [Editor] Change movement speed with MouseWheel

* Change: Use Mousewheel during movement to control the movement speed of the camera

* [Editor] Use Camera movementspeed methods to control

* [Editor] Improve speed increments

* [Editor] Fix camera panning speed

* Change: Do not use basespeed multiplier, to be independent from current movement speed multiplier

* [Editor] Improve panning & Zoomlevels

* [Editor] Improve Comments

* [Editor] Improve speed calc, speedsteps & UI-layout

* Change: Calculate panningspeed by multiplication & predefine the factor
* Change: Add more speed increments
* Change: Improve Layout in Sceneview -> Camerasettings (add display of current speed-value and extended sliderbar size, due to more increments)

* [Editor] Fix margins for cameraspeed settings

VaclavElias committed:
[XML Comment] Minor updates on EFlags in CollisionFilterGroups.cs (#1910)

acastrodev committed:
[Editor] Fixes projects disappearing from the launcher after exiting without saving (#1880)

This commit addresses the bug reported where projects were not persisting correctly in the launcher after unsaved exits of saved projects.

    - The EditorViewModel's RemoveRecentFile method calls the RemoveFile method in the MRUDictionary.
    - This triggers the execution of mostRecentlyUsedFiles.RemoveWhere(x => string.Equals(x.FilePath, filePath, StringComparison.OrdinalIgnoreCase)).
    - Similarly, when the AddFile method is used in the MRUDictionary, it also executes the same RemoveWhere operation.

Refs: #1184

Ethereal77 committed:
[Build] Fix typo in translations generation (#1916)

* Also format the target a bit to aid in legibility

adrsch committed:
[Import] Handle importing meshes with duplicate material names (#1920)

Fixes an error I ran into importing a FBX that was created with 3ds Max 2023.

Jklawreszuk committed:
[Native] Move C++ methods for SpriteBatch and FastTextRenderer to C# (#1896)

Eideren committed:
[Launcher] Prevent launcher automatically closing when the user does not have internet access (#1912)

Jklawreszuk committed:
[Editor] Fix warnings, fix crash report logo, minor refactor (#1741)

Jklawreszuk committed:
[Build] Remove MSBuild.Extras from project (#1895)

Eideren committed:
[Editor] Allow drag and drop of EntityComponent (#1921)

froce committed:
Address PR feedback

adrsch committed:
[Editor] Add settings for the camera speed increase/decrease hotkeys (#1927)

froce committed:
Add tests and improve comments.

froce committed:
Add GameProfilingSorting.ByAverageTime

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

froce committed:
Revert premature pessimization

froce committed:
Prevent division by 0 in GameProfilingSystem

froce committed:
Minor ProfilingKey cleanup

adrsch committed:
[Physics] Let the user set the default Bullet gravity vector in PhysicsSettings (#1928)

Jklawreszuk committed:
[Tools] Migrate Irony.GrammarExplorer to net 6.0 (#1932)

* Fix Irony.GrammarExplorer formating using `dotnet format`

* Irony.GrammarExplorer - Fix few warnings

* Irony.GrammarExplorer - Apply some fixes from codeplex repo

manio143 committed:
Merge pull request #1788 from froce/trace-profiling

[Core] Enable multiple profiler consumers and add a timeline/tracing profiler

fydar committed:
[Build] Fixed an errors in the build pipeline associated with having a space in the user name (#1941)

* [Build] Fixed an exit code 9009 encountered during the build pipeline

* [Build] Fixed an exit code 123 encountered during the build pipeline

---------

Co-authored-by: Eideren <contact@eideren.com>

froce committed:
fix(graphics): Stop FastTextRenderer VB clobbering (#1954)

`FastTextRenderer` was overwriting the start of the vertex buffer for every string. This change advances the start of the span to prevent that.

WojciechNagorski committed:
Update SSH.NET to 2023.0.0 (#1951)

froce committed:
[Android] Fix Android build error (#1949)

fydar committed:
[Docs] Use XML documentation lists (#1948)

Kryptos-FR committed:
[Editor] Remove some windows dependencies in editor libraries (#1908)

VaclavElias committed:
Xml comments fixing 1 (#1918)

* [XML Comment] return value corrected to Graphics.PixelBuffer

* fix: Typo and grammar corrected

* [XML Comment] cref requiredFields removed because it is private variable, hidden from docs

* [XML Comment] crefs corrected

* [XML Comment] Lower case cref corrected to upper case

* [XML Comment] cref sound corrected to Sound

* [XML Comment] corrected

* fix: Typo fixed in the xml comments

* [XML Comment] cref removed, namespace collision

* [XML Comment] coments improvemnets

* [XML Comment] cref removed, because it is private

* [XML Comment] cref removed, not accessible

* [XML Comment] cref removed, not accessible

* [XML Comment] cref corrected

* [XML Comment] typos and cref corrections

* [XML Comment] cref corrected

* [XML Comment] cref removed, the methods are not visible

* [XML Comment] cref updated with TextureDimension

* Update Image.cs - Space removed

Kryptos-FR committed:
[Presentation] Reduce allocations when parsing number in NumericTextBox (#1955)

Eideren committed:
[Sample] Replace deprecated GetServiceAs calls (#1943)

Basewq committed:
[Assets] Fix the CompilerApp being triggered per Runtime Identifiers due to Microsoft.Android.Sdk executing sub-builds (fixes #1901). (#1905)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

JeromyWalsh committed:
Removed all references to $(SolutionDir) from build (#1894)

Removed all references to SolutionDir or replaced with references to $(MSBuildThisFileDirectory).

Co-authored-by: Eideren <contact@eideren.com>

Eideren committed:
[Serialization] Fix diverging rules for editor and runtime serialization of fields and properties (#1875)

acastrodev committed:
feat(extension): Rename launcher buttons for clarity (#1872)

* feat(extension): Rename launcher buttons for clarity

   - Update launcher extension button labels
   - New labels: "Visual Studio Shader addons"
   - Add link under VisualStudioExtension label: "An installed Visual Studio is required, which can be found here"

This commit addresses the issue of misleading launcher extension button labels, making it clearer that users are installing an extension, not the IDE itself. The new labels and added clarification help redirect users to the Visual Studio download page to avoid confusion.

* Refactor LauncherWindow.xaml.cs and LauncherWindow.xaml

In LauncherWindow.xaml.cs:
- Added a Loaded event handler method TextBlockVisualStudioDownloadPage_Loaded to handle the loading of the TextBlock displaying Visual Studio download information.
- Implemented logic to check for compatible Visual Studio versions and hide the TextBlock if compatible versions are found.

In LauncherWindow.xaml:
- Added the Loaded event handler TextBlockVisualStudioDownloadPage_Loaded to the TextBlock element responsible for displaying Visual Studio download information.

This change improves the user experience by dynamically hiding the Visual Studio download information TextBlock when compatible versions are detected, providing a cleaner interface.

* Update capitalizaton in "Visual Studio {0} extension"

---------

Co-authored-by: Alexandre Correa Castro <alexandre.castro@hotmail.com.br>

IXLLEGACYIXL committed:
Stride Diagnostics Analyzer (#1864)

* Set up analyzer project to be included in every project

* Create basic structure

* Add Tests project

* Add STRDIAG003

* add STRDIAG004

* add STRDIAG005

* add STRDIAG006

* cleanup with wellknownreferences

* cleanup code

* add strdiag007

* format

* empty line removed

* expand references

* add attributehelper

* addSTRDIAG008

* add strdiag000

* repair assign mode

* repair STRDIAG000

* fix strdiag000

* fix strdiag008

* improve strdiag004

* finally fixed with originaldefinition

* Performance update

* fix delegate check

* fix delegate comparison on delegate

* add comment for Updatable attribute

* make HasAttribute an extension method and remove attributehelper

* change strdiag003 error message

* update rules for STRDIAG004

* rename strdiag005

* change strdiag005 error message

* rename to Serialization category

* refactor AnalyzeField

* remove useless null checks

* rename class to match file name

* fix error messages

* remove try catch

* update error message

* update delegate error message

* add missing accessibility check

* add missing accesibility check

* add missing accesibility check

* make accessibility check extension method

* fix readonly member, negate isvisible

* fix title

* cleanup ifs

* project cleanup

* reorganize if statements

* extract if

* fix spacing

* remove unused parameter

* fix style

* remove unused paramter

* add some tests

* rename files

* more tests

* add tests, fix typos

* add todo

* rename tests

* cleanup

* addd invalid content mode

* cleanup analyzers

* cleanup tests

* fix tests and bugs

* update test

* add edge cases, update tests, cleanup

* add rest of the tests

* add updatable test

* update comments

* make repot diagnostics an extension method

* add immutable comment

* compiler utils cleanup

* format cleanup

* upgrade to latest roslyn

* remove folder

* remove visibility , add using directive for debugger

* add reason

* renamed template

* remove String

* rename tests

* named parameter, bump version of test project

* add comments

* fix and strdiag001

* add strdiag009

* add strdiag001 tests

* make class proper accessor for serialization rules

* alter message

* add pragma disable

* update message

* add pragma

* fix test

* add strdiag009 tests

* add enums to supported values

* convert title to fit to datacontract title

* add links

* add link

* add missing help link

---------

Co-authored-by: Marian Dziubiak <marian.dziubiak@gmail.com>
Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

MaximilianEmel committed:
removes /deps/ (was already in variable) (#1956)

ch3mbot committed:
[Math] Add a couple of helpers for Vectors (#1769)

Kryptos-FR committed:
[Math] Fix #1769 and introduce an optional argument to specify a different rounding strategy if needed. (#1964)

Eideren committed:
[Github] Update pull request template to ensure users tried out their changes (#1965)

Doprez committed:
[Math] fixed docs typo (#1976)

Eideren committed:
Revert "[Editor] Remove some windows dependencies in editor libraries (#1908)" (#1980)

This reverts commit 329becbecf0c392b37feb5c8bc6007bd4ee2f567.

Eideren committed:
[Docs] Move bounty paragraph to a more prominent spot

Basewq committed:
[AssemblyProcessor] Fixed packing path.

manio143 committed:
Merge pull request #1987 from Basewq/asm_fix

[AssemblyProcessor] Fixed packing path.

IXLLEGACYIXL committed:
[Core] Make object id more performant (#1957)

* make object id span

* string.create todo

* new hashcode

* remove comment

* remove empty line

* conversion improvements

* Make it work with AssemblyProcessor

* Format

---------

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>
Co-authored-by: Marian Dziubiak <marian.dziubiak@gmail.com>

tebjan committed:
[Readme] Some additional info for building Stride from source

azeno committed:
[Shaders] Fixes `EffectValueDescription.DefaultValue` for negative values

Negative values are represented as unary expressions. This commit adds support for unary expressions when reading the default value of a parameter.
In case an initializer expression is not supported / can't be evaluated the `DefaultValue` property is set to `null`.

Fixes issue #1963

Eideren committed:
[Docs] Update PropertiesDemo.cs

Jklawreszuk committed:
Fix failing to load index on non-Win desktop platforms

Basewq committed:
[Build] Fix native library loading picking up invalid files (fixes #1750)

Eideren committed:
[Editor] Re-introduce workaround for missing input while navigating (#1897)

* [Editor] Re-introduce workaround for missing input while navigating

* [Editor] Slightly faster transition between scene views

Eideren commented:
Sure, I'll change this to a draft then

Eideren committed:
[VR] Remove framecap from VR sample

manio143 committed:
Merge pull request #1999 from Basewq/font_fix

[Build] Fix native library loading picking up invalid files

MaximilianEmel committed:
Fixes OpenXR (#1911)

* Change version of Silk.NET.OpenXR from 2.15.0 to 2.17.1

* using vrFullSurface for Mirror for now

* Improve some Exception and Logging text

* Use XR_REFERENCE_SPACE_TYPE_STAGE as default because standing VR is generally the standard nowadays, rather than seated.

* Deleted deprecated Fove and GoogleVR VirtualReality APIs

* Completely remove Fove & GoogleVR

* Changes OpenVR's default Universe to Standing.

* Set default Universe for OpenVR & OpenXR back to Seated.

* Update all (Stride.Graphics) uses of Silk.NET from 2.15.0 to 2.17.1

* Update System.Memory from 4.5.4 to 4.5.5 to match Silk.NET

fydar committed:
[Breaking] Scoping generic extension methods (#1959)

This was causing every object to have additional methods appearing in the IDE and on the documentation, which can be very confusing for new developers and irritating for experienced ones.

This means that these extension methods no longer work with IntPtr and "IReferencable" types, but this feature was unused.

Seperate extension methods for "IReferencable" types could be created without "Dispose" in the name and instead use the "Release" naming.

* Adding ReleaseBy and RemoveReleaseBy and fixed documentation comments

Adding ReleaseBy and RemoveReleaseBy to replace the lost functionality of DisposeBy.

Fixed typos and overly-aggressive copy pasting in documentation comments.

Jklawreszuk committed:
Add information about Iron.GrammarExplorer project

manio143 committed:
Merge pull request #2007 from Jklawreszuk/Jklawreszuk-patch-1

Add information about Irony.GrammarExplorer project

manio143 committed:
Remove empty file references from Stride.Assets

manio143 commented on sources/engine/Stride.Assets/Stride.Assets.csproj:
```diff
@@ -65,25 +65,6 @@
       <Link>Properties\SharedAssemblyInfo.cs</Link>
     </Compile>
   </ItemGroup>
-  <ItemGroup>
```
Removing those `<None Include` because it references files under Stride.Engine and Stride.UI - blame shows this is a remnant from the initial commit.
I'm guessing those were added to allow MSBuild to track changes in assets files and rebuild the project when one of them changes. Currently that is not in place elsewhere and changing assets requires manually touching any C# related files to trigger MSBuild to rebuild the project.

manio143 commented:
I just noticed that for example GameProfilingSystem in Stride.Engine takes a runtime dependency on the default sprite font asset. With this PR I'd break it if someone imports Stride.Engine but doesn't import Stride.UI

Ethereal77 commented:
Is `GameProfilingSystem` not using `DebugText` instead of the regular `SpriteFont`? If using `DebugText`, it is loading the Debug Font sprite sheet, which is not a "standard" sprite font asset. With this I mean that even if we move the default sprite font asset to other package, if we keep the debug font in `Stride.Engine` it should work. Shouldn't it?

manio143 commented:
It uses the FastTextRender rather than DebugText. See here:
https://github.com/stride3d/stride/blob/ffd71acc8a8a26a80e5275f833b6e1442e98d01c/sources/engine/Stride.Engine/Profiling/GameProfilingSystem.cs#L354-L358

Ethereal77 commented:
> It uses the FastTextRender rather than DebugText. See here:
> 
> https://github.com/stride3d/stride/blob/ffd71acc8a8a26a80e5275f833b6e1442e98d01c/sources/engine/Stride.Engine/Profiling/GameProfilingSystem.cs#L354-L358

Ok, that's what I meant. It is not a `SpriteFont` but a regular `Texture` that `FastTextRenderer` knows how to use. So that texture should stay in `Stride.Engine` as it is a "private" spritefont-like asset, not something the user can/should use.
